tests/models/test_skipmixin.py

```diff
@@ -117,3 +117,51 @@ def get_state(ti):
 
         assert get_state(ti2) == State.NONE
         assert get_state(ti3) == State.SKIPPED
+
+    def test_skip_all_except_none(self, dag_maker):
```

Comment:
You can avoid all these new tests (which duplicate most code in `test_skip_all_except`) with `pytest.mark.parametrize`.